### ISSUE TITLE
Add a field to notes to indicate what software is creating them

### ISSUE THREAD

**Issue by ToeBee** [openstreetmap/openstreetmap-website#385]
As more apps and external  entities like Craigslist start using our "Notes" feature, it would be helpful to see where notes are coming from. 

For example if they come from Craigslist, I know that the user was looking at a custom rendering that may be several weeks out of date. If it is from a smartphone app it means the user was probably in the field and had eyes on the situation when the note was created. If it was created in JOSM then it is likely an armchair mapper making a note of something he didn't have time to fix up while doing some other mapping.

Basically I want a "created_by" tag for notes just like changesets have now. Obviously notes don't have tags and I don't see much use in allowing arbitrary tagging on notes so it should just be its own field.

**Comment by westnordost** [comment]
If this is simply about identifying with which user agent the user created or commented the note, why not simply use the UserAgent from the HTTP header?

**Comment by peternewman** [comment]
> If this is simply about identifying with which user agent the user created or commented the note, why not simply use the UserAgent from the HTTP header?

That wouldn't differentiate between Craigslist and the OSM website and the OSM website on a browser in the field though would it, unless you also included the referrer or something?

**Comment by andrewharvey** [comment]
This would help with https://github.com/streetcomplete/StreetComplete/issues/2692#issuecomment-850715460 but in general it's very useful for mappers to know if the note came from a survey app, or website. Right now some apps embed this content within the description
